### DISCUSSION TITLE
Separate UI and code

### DISCUSSION THREAD

**Discussion by NelsonLamprecht** [solidjs/solid#891] [Q&A] [answered]
One of the reasons I prefer Angular over other UI libraries is the separation of HTML from the code. 
Obviously this requires some type of update cycles to ensure things are in sync.
Is this possible with SolidJS?

**Comment by ghost** [comment, marked as answer]
You can make Store files that contains State & it's logic, because Solid.js can handle global State, You can import & use it in Your Jsx/Tsx files. I mostly Prefer to use Global States instead of Passing Props to child Components. But  You can't Separate Your Component in to HTML, CSS, TS files as you used to in Angular.

Just see these Pi Guessing Simple App: 
https://github.com/ZeAmanuel28/pi-guesser

**Reply by NelsonLamprecht** [reply]
Thank you. I really feel the separation of UI and logic is important when working in a team of developers especially when that team consists of designers and developers.

**Reply by ghost** [reply]
Yes, It's useful & easy to implement in Solid.js, Because Solid.js is Reactive, Performant it's simple to separate UI & State. If You just like Single File Components, Solid.js has Solid SFC (https://github.com/LXSMNSYC/solid-sfc) (https://github.com/LXSMNSYC/solid-sfc-styled-labels-starter) & Also Solid Labels (https://github.com/LXSMNSYC/solid-labels).
